src/cpu-info.ts

```diff
@@ -1,18 +1,13 @@
-export type CpuInfo = {
-  numLogicalCores?: number;
-};
-
 /**
- * Get the available information about the machine's CPU.
- *
- * @returns The {@link CpuInfo}.
+ * A class that provides information about the CPU.
  */
-export function getCpuInfo(): CpuInfo {
-  const cpuInfo = {} as CpuInfo;
-
-  if (navigator.hardwareConcurrency) {
-    cpuInfo.numLogicalCores = navigator.hardwareConcurrency;
+export class CpuInfo {
+  /**
+   * Gets the number of logical CPU cores.
+   *
+   * @returns The number of logical CPU cores, or undefined if not available.
+   */
+  static getNumLogicalCores(): number {
```

Comment:
I noticed that the JSDoc says this method could return "undefined if not available" but here we say it always returns a number. [Checking the MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/hardwareConcurrency) it does seem like all major browsers support it, though I suppose it is possible that someone might use an unsupported browser.

Should we set the return type to `number | undefined` here?

Reply:
yes, that was my initial idea as well. I checked the MDN docs, but when we return `number | undefined` I am getting an error for:

<img width="760" alt="Screenshot 2024-05-22 at 11 04 52 AM" src="https://github.com/webex/web-capabilities/assets/67634227/451fb4b4-3bb3-4eba-b620-2bd2705338da">

 We could avoid this by using `!`, but we've already agreed to not use it.

Reply:
Another option could be using something like:

```
static isCapableOfSending1080pVideo(): CapabilityState {
    const numCores = CpuInfo.getNumLogicalCores();
    if (numCores === undefined) {
      return CapabilityState.UNKNOWN;
    }
    if (numCores < 2) {
      return CapabilityState.NOT_CAPABLE;
    }
    return CapabilityState.CAPABLE;
  }
  ```